### ISSUE TITLE
Different Translation on Yandex

### ISSUE THREAD

**Issue by benjieperez** [Animenosekai/translate#79]
BTW Thanks for this library its such a nice library and suites to my on going project.

My question is I found out testing Yandex Translate on Live vs The Library Itself and it shows a different translation.

This is the text.

{
    'original_text': 'And as surely as that, we will achieve the country all Filipinos deserve. God bless the Philippines, God bless our work. Maraming, maraming salamat po sa inyong lahat!',
    'translated_text': 'Und so sicher werden wir das Land erreichen, das alle Filipinos verdienen. Gott segne die Philippinen, Gott segne unsere Arbeit. Marschieren, marschieren salamat po sa inyong lahat!'
  }
  
  
![image](https://user-images.githubusercontent.com/20369043/224614040-1f08801c-547b-4784-842e-0c5578423189.png)

But when I segment the full text into single sentences. It actually translate the same.

**Comment by ZhymabekRoman** [comment]
That's strange, have you tried the same in the mobile application Yandex Translate? `translatepy` uses Android backend API.

**Comment by benjieperez** [comment]
Yeah, I try the Yandex Translate using translatepy directly and it outputs the 'Und so sicher werden wir das Land erreichen, das alle Filipinos verdienen. Gott segne die Philippinen, Gott segne unsere Arbeit. Marschieren, marschieren salamat po sa inyong lahat!'. Different to the Yandex Translate on Browser.

It only occurs if the string is multilingual.

**Comment by benjieperez** [comment]
I used a different user agent in yandex.py and it actually give me the right translation.

![image](https://user-images.githubusercontent.com/20369043/224733132-95d2822a-5621-42d5-8e3a-2e401a439f85.png)
![image](https://user-images.githubusercontent.com/20369043/224733251-109992c5-f26a-4e03-ba52-85898ea515b7.png)

**Comment by benjieperez** [comment]
Maybe a user agent randomizer will fix on this?

**Comment by benjieperez** [comment]
I have this simple user agent randomizer.

f'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.{random.randint(0, 9999)} Safari/537.{random.randint(0, 99)}'

Let me know if this would help, Thanks!

**Comment by Animenosekai** [comment]
woooow why didn't I get this notification sooner...

> @Animenosekai, Can we rewrite the current user agent generation algorithm with a random user agent data information combination (OS, browser, version and etc.) instead of just getting a random user agent string in [useragents](https://github.com/Animenosekai/useragents)?

Yup, this seems totally doable !

I don't have much time now, but whenever I get the time to 🍡

**Comment by ZhymabekRoman** [comment]
So I did some research and came up with some interesting results. Original text: `And as surely as that, we will achieve the country all Filipinos deserve. God bless the Philippines, God bless our work. Maraming, maraming salamat po sa inyong lahat!` is in Tagalog. Translatepy recognises it as English. And the web version of Yandex Translate also detects it as English in the first lookup. In a second lookup Yandex web partially performs text language detection, in our case it's sending `salamat po sa inyong lahat!` to detect language, and it's returned as `Tagalog`. User-agent there is no common sense.

**Comment by Animenosekai** [comment]
> So I did some research and came up with some interesting results. Original text: `And as surely as that, we will achieve the country all Filipinos deserve. God bless the Philippines, God bless our work. Maraming, maraming salamat po sa inyong lahat!` is in Tagalog. Translatepy recognises it as English. And the web version of Yandex Translate also detects it as English in the first lookup. In a second lookup Yandex web partially performs text language detection, in our case it's sending `salamat po sa inyong lahat!` to detect language, and it's returned as `Tagalog`. User-agent there is no common sense.

So they have a 2-pass system for language detection where they check the language a second time ??

**Comment by ZhymabekRoman** [comment]
> So they have a 2-pass system for language detection where they check the language a second time ??

Yes, I just checked the Android application, it also correctly detects the language of the text as Tagalog.

**Comment by ZhymabekRoman** [comment]
Today, Yandex incorrectly detects the language of text. So `translatepy` has nothing to do with it:
![image](https://github.com/Animenosekai/translate/assets/61125068/c5e03491-2384-4919-8730-da445cf531e5)
